### PR TITLE
Add packet dropping functionality

### PR DESCRIPTION
**Pull request by BrytniR**
In the simulation software, drop packets with probability specified by -ul or -dl

### CODE REVIEW

File: obc_transceiver_simulator/obc_transceiver_simulator.py [hunk 1]
```diff
@@ -167,11 +167,17 @@ def main_loop():
             metavar=('uart'), help='UART port on programmer')
     parser.add_argument('-b', '--baud', required=False, default=9600,
             metavar=('baud'), help='Baud rate (e.g. 1200, 9600, 19200, 115200')
+    parser.add_argument('-ul', '--uplink-drop', required=False, default=0,
+            metavar=('uplink'), help='Package drop rate from ground to satellite (0-1)')
+    parser.add_argument('-dl', '--downlink-drop', required=False, default=0,
+            metavar=('downlink'), help='Package drop rate from satellite to ground (0-1)')
```
Comment (brunofalmeida) [inline obc_transceiver_simulator/obc_transceiver_simulator.py]: Use the short forms `-ud` and `-dd` instead

File: obc_transceiver_simulator/packets.py [hunk 2]
```diff
@@ -56,6 +58,10 @@ def receive_rx_packet():
 
         # Need 2 <CR> bytes to start/end message
         if len(cr_indices) >= 2:
+            # Drop packet?
+            if (random.uniform(0,1) < Global.downlink_drop):
+                print("Downlink Packet Dropped")
+                return None
```
Comment (brunofalmeida) [inline obc_transceiver_simulator/packets.py]: It's the right logic but if you do it at this point in the code, you don't know if it actually detected a packet with the correct format - should move this to line 82.

File: obc_transceiver_simulator/packets.py [hunk 3]
```diff
@@ -84,7 +90,11 @@ def receive_rx_packet():
 # Use `bytearray` instead of `bytes`
 def send_raw_uart(uart_bytes):
     print("Sending UART (%d bytes):" % len(uart_bytes), bytes_to_string(uart_bytes))
-    Global.serial.write(uart_bytes)
+    # Drop packets?
+    if (random.uniform(0,1) < Global.uplink_drop):
+        print("Uplink Packet Dropped")
+    else:
+        Global.serial.write(uart_bytes)
```
Comment (brunofalmeida) [inline obc_transceiver_simulator/packets.py]: Move this logic to line 125 (in `send_tx_packet()`)

**Review by brunofalmeida — CHANGES_REQUESTED** [review]

File: obc_transceiver_simulator/packets.py (hunk 3 above)
Comment (BrytniR) [reply obc_transceiver_simulator/packets.py]: Should I also put this in the 'Enter raw hex for UART:' command (f) as well?

Comment (brunofalmeida) [reply obc_transceiver_simulator/packets.py]: No, I intended for the raw UART functionality to just always output whatever you tell it.

**Pull request merged**